### DISCUSSION TITLE
Time to prioritize bugs over features?

### DISCUSSION THREAD

**Discussion by jcbritobr** [actix/actix-web#1959] [General]
Actix is going very instable with crescent number of bugs. Is not time to stop features and prioritize stability by fixing issues?

**Comment by fakeshadow** [comment]
What do you mean by very instable? You can file issue for detailed bug report. 
How do you observe feature is prioritize over bug fixing? Like what feature is not useful and cause bugs or instablity you are talking about?

The current priortize of actix-web is to upgrade to tokio 1.0. Not feature nor bug fixes.

**Reply by jcbritobr** [reply]
I mean we are accumulating bugs that are a long time without been fixed. The project are losing stability. I think prioritize an "upgrade" without prioritize bug fix is not a good idea for the project stability.

There are bugs from 2019 like this https://github.com/actix/actix-web/issues/1123 and most of contributions are "improvments", not bug fixes. Well, someone needs to start talking about this. :)

**Reply by fakeshadow** [reply]
Can you scroll down a bit on that issue? You would find there is already PR for fixing it.
Please file an issue for what bugs you are facing. Just go over the issue list and comes up with "instability" is not helping anyone.

**Reply by jcbritobr** [reply]
Why it( #1123) is not closed? I think be reluctant to discuss this doesn't makes sense. There are plenty other long time bugs without solution also.

**Reply by fakeshadow** [reply]
PRs have to be reviewed before it can be merged. It would be closed when the fix is merged to master branch.

You bring that issue specificly and then you say it does not make sense. Sorry but you really lost me.

Please file an issue for the bugs you are encountering. I'm trying to help you.

**Reply by jcbritobr** [reply]
Well its a lot of time without merging fixes, and I also started this thread. Thats are on table to talk, share and think about.

**Reply by fakeshadow** [reply]
Because Nobody is reviewing it. 

All the contributiors in actix-web project are voluenteers and there is no forced work schedule on anyone. Things gonna be slow when people are busy and/or focusing on other stuff. So please be patient.

That said I would ping them for a review on it and get it merge sooner.

**Comment by robjtede** [comment]
You should see that there has been a lot of effort recently in the actix-net repo where lots of the underlying low-level functionality lives. We've been fixing lots of bugs and generally reducing API surface so that our crates can remain without breaking changes for a longer period of time.

Attention will soon turn back to this repo to prepare for the v4 release. We're not going to rush. Bugs will be addressed.
